packages/openapi-generator/src/parser/refs.ts

```diff
@@ -15,3 +19,81 @@ export function resolveObject<T>(
 ): T {
   return isReferenceObject(obj) ? refs.get(obj.$ref) : obj;
 }
+
+export async function createRefs(
+  document: OpenAPIV3.Document
+): Promise<OpenApiDocumentRefs> {
+  const $refs = await resolve(document);
+  return new OpenApiDocumentRefs($refs, parseSchemaRefMapping(document));
+}
+
+export class OpenApiDocumentRefs {
+  constructor(
+    private refs: $Refs,
+    private schemaRefMapping: SchemaRefMapping
+  ) {}
+
+  /**
+   * Check whether the given object is a reference object and resolve if necessary.
+   * This operates only on the current level and does not resolve the object recursively.
+   * @param obj Object to resolve if necessary.
+   * @returns A resolved object.
+   */
+  resolveObject<T>(obj: T | OpenAPIV3.ReferenceObject): T {
+    return isReferenceObject(obj) ? this.refs.get(obj.$ref) : obj;
+  }
+
+  /**
+   * Parse the type name of a reference object.
+   * @param obj Reference object or reference path to get the type name from.
+   * @returns Parsed type name.
+   */
+  getSchemaNaming(obj: OpenAPIV3.ReferenceObject | string): SchemaNaming {
+    const ref = isReferenceObject(obj) ? obj.$ref : obj;
+
+    if (!ref.startsWith('#/components/schemas')) {
+      throw new Error(
+        `Could not get schema naming for reference path '${ref}'. Path does not reference a schema.`
+      );
+    }
+
+    const schemaNaming = this.schemaRefMapping[ref];
+    if (!schemaNaming) {
+      throw new Error(
+        `Could not find schema naming for reference path '${ref}'. Schema does not exist.`
+      );
+    }
+    return schemaNaming;
+  }
+}
+
+function parseSchemaRefMapping(document: OpenAPIV3.Document): SchemaRefMapping {
```

Comment:
Also matter of taste, I would put this as a private method in the class and call this in the constructor.

Reply:
done